diff -up arpwatch-3.1/Makefile.in.4~ arpwatch-3.1/Makefile.in
--- arpwatch-3.1/Makefile.in.4~	2021-01-01 21:46:22.461085912 +0100
+++ arpwatch-3.1/Makefile.in	2021-01-01 21:47:21.561918513 +0100
@@ -67,6 +67,7 @@ LDFLAGS = @LDFLAGS@
 LIBS = @LIBS@
 # Standard LIBS without libpcap.a
 SLIBS = @LBL_LIBS@
+LDFLAGS =
 
 INSTALL = @INSTALL@
 CPPCHECK = cppcheck -q --enable=all --inline-suppr --suppress=variableScope -Dlint -D__FreeBSD__ -I/usr/include
@@ -109,11 +110,11 @@ all: $(ALL)
 
 $(PROG): $(WOBJ) @V_PCAPDEP@
 	@rm -f $@
-	$(CC) $(LDFLAGS) -o $@ $(WOBJ) $(LIBS)
+	$(CC) $(LDFLAGS) -o $@ $(WOBJ) $(LDFLAGS) $(LIBS)
 
 arpsnmp: $(SOBJ)
 	@rm -f $@
-	$(CC) $(LDFLAGS) -o $@ $(SOBJ) $(SLIBS)
+	$(CC) $(LDFLAGS) -o $@ $(SOBJ) $(LDFLAGS) $(SLIBS)
 
 purify: $(SOBJ)
 	@rm -f $@
