diff -up arpwatch-3.1/arpwatch.c.2~ arpwatch-3.1/arpwatch.c
--- arpwatch-3.1/arpwatch.c.2~	2019-11-30 19:35:23.000000000 +0100
+++ arpwatch-3.1/arpwatch.c	2021-01-01 21:40:49.739325396 +0100
@@ -71,6 +71,7 @@ struct rtentry;
 #include <string.h>
 #include <syslog.h>
 #include <unistd.h>
+#include <pwd.h>
 
 #include <pcap.h>
 
@@ -170,6 +171,26 @@ int	sanity_fddi(struct fddi_header *, st
 int	toskip(u_int32_t);
 void	usage(void) __attribute__((noreturn));
 
+
+void dropprivileges(const char* user)
+{
+	struct passwd* pw;
+	pw = getpwnam( user );
+	if ( pw ) {
+		if ( initgroups(pw->pw_name, NULL) != 0 || setgid(pw->pw_gid) != 0 ||
+				 setuid(pw->pw_uid) != 0 ) {
+			syslog(LOG_ERR, "Couldn't change to '%.32s' uid=%d gid=%d", user,
+						 pw->pw_uid, pw->pw_gid);
+			exit(1);
+		}
+	}
+	else {
+		syslog(LOG_ERR, "Couldn't find user '%.32s' in /etc/passwd", user);
+		exit(1);
+	}
+	syslog(LOG_DEBUG, "Running as uid=%d gid=%d", getuid(), getgid());
+}
+
 int
 main(int argc, char **argv)
 {
@@ -181,6 +202,7 @@ main(int argc, char **argv)
 	char *interface, *rfilename;
 	struct bpf_program code;
 	char errbuf[PCAP_ERRBUF_SIZE];
+	char *serveruser = NULL;
 
 	if (argv[0] == NULL)
 		prog = "arpwatch";
@@ -198,7 +220,7 @@ main(int argc, char **argv)
 	interface = NULL;
 	rfilename = NULL;
 	pd = NULL;
-	while ((op = getopt(argc, argv, "CdD:Ff:i:n:NpP:qr:svw:W:x:zZ")) != EOF)
+	while ((op = getopt(argc, argv, "CdD:Ff:i:n:NpP:qr:su:vw:W:x:zZ")) != EOF)
 		switch (op) {
 
 		case 'C':
@@ -258,6 +280,15 @@ main(int argc, char **argv)
 			++suppress;
 			break;
 
+		case 'u':
+			if (optarg) {
+				serveruser = strdup(optarg);
+			} else {
+				fprintf(stderr, "%s: Need username after -u\n", prog);
+				usage();
+			}
+			break;
+
 		case 'v':
 			++vrrpflag;
 			break;
@@ -383,8 +414,11 @@ main(int argc, char **argv)
 	 * Revert to non-privileged user after opening sockets
 	 * (not needed on most systems).
 	 */
-	setgid(getgid());
-	setuid(getuid());
+	/*setgid(getgid());*/
+	/*setuid(getuid());*/
+	if ( serveruser ) {
+		dropprivileges( serveruser );
+	}
 
 	/* Must be ethernet or fddi */
 	linktype = pcap_datalink(pd);
@@ -927,6 +961,6 @@ usage(void)
 	    "usage: %s [-CdFNpqsvzZ] [-D arpdir] [-f datafile]"
 	    " [-i interface]\n\t"
 	    " [-P pidfile] [-w watcher@email] [-W watchee@email]\n\t"
-	    " [-n net[/width]] [-x net[/width]] [-r file]\n", prog);
+	    " [-n net[/width]] [-x net[/width]] [-r file] [-u username]\n", prog);
 	exit(1);
 }
