diff -up arpwatch-3.1/Makefile.in.1~ arpwatch-3.1/Makefile.in
--- arpwatch-3.1/Makefile.in.1~	2021-01-01 21:33:15.730468895 +0100
+++ arpwatch-3.1/Makefile.in	2021-01-01 21:33:53.435063354 +0100
@@ -125,8 +125,8 @@ version.c: $(srcdir)/VERSION
 	sed -e 's/.*/const char version[] = "&";/' $(srcdir)/VERSION > $@
 
 install: $(ALL)
-	$(INSTALL) -v -m 555 $(PROG) $(DESTDIR)$(BINDEST)
-	$(INSTALL) -v -m 555 arpsnmp $(DESTDIR)$(BINDEST)
+	$(INSTALL) -v -m 755 $(PROG) $(DESTDIR)$(BINDEST)
+	$(INSTALL) -v -m 755 arpsnmp $(DESTDIR)$(BINDEST)
 	@diff $(srcdir)/$(PROG).8 $(DESTDIR)$(MANDEST)/man8 >/dev/null 2>&1 || \
 	    $(INSTALL) -v -m 444 $(srcdir)/$(PROG).8 $(DESTDIR)$(MANDEST)/man8
 	@diff $(srcdir)/arpsnmp.8 $(DESTDIR)$(MANDEST)/man8 >/dev/null 2>&1 || \
